swarm_copy/app/dependencies.py

```diff
@@ -93,12 +118,78 @@ def get_starting_agent(
         instructions="""You are a helpful assistant helping scientists with neuro-scientific questions.
                 You must always specify in your answers from which brain regions the information is extracted.
                 Do no blindly repeat the brain region requested by the user, use the output of the tools instead.""",
-        tools=[PrintAccountDetailsTool],
+        tools=[
+            LiteratureSearchTool,
+            ElectrophysTool,
+            GetMEModelTool,
+            GetMorphoTool,
+            KGMorphoFeatureTool,
+            MorphologyFeatureTool,
+            ResolveEntitiesTool,
+            GetTracesTool,
+        ],
         model=settings.openai.model,
     )
     return agent
 
 
+async def get_httpx_client(request: Request) -> AsyncIterator[AsyncClient]:
+    """Manage the httpx client for the request."""
+    client = None
+    try:
+        client = AsyncClient(
+            timeout=None,
+            verify=False,
+            headers={"x-request-id": request.headers["x-request-id"]},
+        )
+        yield client
+    finally:
+        if client is not None:
+            await client.aclose()
```

Comment:
`AsyncClient` should never raise an exception at instantiation. even if it did, it would raise an error 500 but having the client as None would likely do the same. I suggest we stick to 
```python
    client = AsyncClient(timeout=settings.metadata.timeout, verify=False, headers={"x-request-id": request.headers["x-request-id"]},)
    try:
        yield client
    finally:
        await client.aclose()
```